### PR TITLE
New: Extracter for Porntrex

### PR DESCRIPTION
**Pull request by nindogo**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

Added [Porntrex](https://www.porntrex.com/). Both Video Page and Playlist page.

### CODE REVIEW

File: youtube_dl/extractor/porntrex.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+)
+
+
+class PornTrexIE(InfoExtractor):
+    '''Class for downloading Porntrex video.'''
+    _NETRC_MACHINE = 'porntrex'
+    _VALID_URL = r'https?://(?:www\.)?porntrex\.com/video/(?P<id>[0-9]+)/.*'
+    _TEST = {
+        'url': 'https://www.porntrex.com/video/519351/\
+                be-ariyana-adin-breaking-and-entering-this-pussy',
```
Comment (dstftw) [inline youtube_dl/extractor/porntrex.py]: Do not split literals.

File: youtube_dl/extractor/porntrex.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+from ..utils import (
+    urlencode_postdata,
+    ExtractorError,
+)
+
+
+class PornTrexIE(InfoExtractor):
+    '''Class for downloading Porntrex video.'''
+    _NETRC_MACHINE = 'porntrex'
+    _VALID_URL = r'https?://(?:www\.)?porntrex\.com/video/(?P<id>[0-9]+)/.*'
```
Comment (dstftw) [inline youtube_dl/extractor/porntrex.py]: `.*` is useless at the end.

File: youtube_dl/extractor/porntrex.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+
+from .common import InfoExtractor
+from ..utils import (
+    urlencode_postdata,
+    ExtractorError,
+)
+
+
+class PornTrexIE(InfoExtractor):
+    '''Class for downloading Porntrex video.'''
```
Comment (dstftw) [inline youtube_dl/extractor/porntrex.py]: Remove pointless comments.

File: youtube_dl/extractor/porntrex.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+            'id': '519351',
+            'ext': 'mp4',
+            'title': 'BE - Ariyana Adin - Breaking And Entering This Pussy',
+            'uploader': 'brand95',
+            'description': 'BE - Ariyana Adin - Breaking And Entering This \
+Pussy',
+        }
+    }
+
+    def get_resolution(self, url):
```
Comment (dstftw) [inline youtube_dl/extractor/porntrex.py]: Inline all such methods and use regexes instead.

File: youtube_dl/extractor/porntrex.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+        formats = []
+        for x, _ in enumerate(url2):
+            formats.append({'url': url2[x],
+                            'ext': url2[x].split('.')[-1],
+                            'resolution': self.get_resolution(url2[x]),
+                            'protocol': self.get_protocol(url2[x]),
+                            })
+        # self.get_thumbnails(webpage)
+        self._sort_formats(formats)
+        print(formats)
```
Comment (dstftw) [inline youtube_dl/extractor/porntrex.py]: Remove all garbage.

**Comment by nindogo** [comment]
Hi,

got this and on it.

File: youtube_dl/extractor/porntrex.py (hunk 1 above)
Comment (nindogo) [reply youtube_dl/extractor/porntrex.py]: Done.
The length of the lines made pylint to complain - but resolved now.

File: youtube_dl/extractor/porntrex.py (hunk 2 above)
Comment (nindogo) [reply youtube_dl/extractor/porntrex.py]: Noted. Thanks for the information.
Removed them

File: youtube_dl/extractor/porntrex.py (hunk 3 above)
Comment (nindogo) [reply youtube_dl/extractor/porntrex.py]: Ok.

File: youtube_dl/extractor/porntrex.py (hunk 5 above)
Comment (nindogo) [reply youtube_dl/extractor/porntrex.py]: Done

**Comment by nindogo** [comment]
Hey @dstftw hope you are doing well.

I would please like to understand why this has been marked as do not update please.
Could you please advise? Can I restart again (deleting this then reissuing the plugin)?

**Comment by dstftw** [comment]
PR should not contain irrelevant changes already existing in master.

**Comment by soccersteve72** [comment]
Any updates on this? I would quite like this extractor...

**Pull request closed without merging**